packages/wallet/backend/src/createContainer.ts

```diff
@@ -289,10 +294,9 @@ export const createContainer = (config: Env): Container<Bindings> => {
     return new RafikiController({ logger, rafikiService, ratesService })
   })
 
-  container.singleton(
-    'quoteService',
-    async () =>
-      new QuoteService({
+  container.singleton('quoteService', async () => {
+    if (!quoteServiceResolved) {
```

Comment:
Imo I don't think this layer of caching services is needed. If you look inside the container class, everything is already kept inside a map, and fetching out of it is syncrhonous in itself. Just the container.resolve method itself is async, in case async dependencies are ever needed.

Same goes for above

Reply:
so this is a circular dependency case?